kernel/kernel-default/src/test/java/io/delta/kernel/parquet/TestParquetBatchReader.java

```diff
@@ -0,0 +1,425 @@
+                    if (expValue == null) {
+                        assertTrue(vector.isNullAt(batchWithIdx._2));
+                    }
+                    else {
+                        assertEquals(expValue.intValue(), vector.getInt(batchWithIdx._2));
+                    }
+                    break;
+                }
+                case "longtype": {
+                    Long expValue = (rowId % 25 != 0) ? rowId + 1L : null;
+                    if (expValue == null) {
+                        assertTrue(vector.isNullAt(batchWithIdx._2));
+                    }
+                    else {
+                        assertEquals(expValue.longValue(), vector.getLong(batchWithIdx._2));
+                    }
+                    break;
+                }
+                case "floattype": {
+                    Float expValue = (rowId % 28 != 0) ? (rowId * 0.234f) : null;
+                    if (expValue == null) {
+                        assertTrue(vector.isNullAt(batchWithIdx._2));
+                    }
+                    else {
+                        assertEquals(expValue.floatValue(), vector.getFloat(batchWithIdx._2), 0.02);
+                    }
+                    break;
+                }
+                case "doubletype": {
+                    Double expValue = (rowId % 54 != 0) ? (rowId * 234234.23d) : null;
+                    if (expValue == null) {
+                        assertTrue(vector.isNullAt(batchWithIdx._2));
+                    }
+                    else {
+                        assertEquals(expValue.doubleValue(), vector.getDouble(batchWithIdx._2),
+                            0.02);
+                    }
+                    break;
+                }
+                case "stringtype": {
+                    String expValue = (rowId % 57 != 0) ? Integer.toString(rowId) : null;
+                    if (expValue == null) {
+                        assertTrue(vector.isNullAt(batchWithIdx._2));
+                    }
+                    else {
+                        assertEquals(expValue, vector.getString(batchWithIdx._2));
+                    }
+                    break;
+                }
+                case "binarytype": {
+                    byte[] expValue = (rowId % 59 != 0) ? Integer.toString(rowId).getBytes() : null;
+                    if (expValue == null) {
+                        assertTrue(vector.isNullAt(batchWithIdx._2));
+                    }
+                    else {
+                        assertArrayEquals(expValue, vector.getBinary(batchWithIdx._2));
+                    }
+                    break;
+                }
+                case "timestamptype": {
+                    throw new UnsupportedOperationException("not yet implemented: " + name);
+                }
+                case "decimal": {
+                    throw new UnsupportedOperationException("not yet implemented: " + name);
+                }
+                case "nested_struct": {
```

Comment:
Test case for null for structs as well? (Or at least future todo?)

Reply:
There are second level structs which are null.